llvm/lib/MC/MCCASObjectV1.cpp

```diff
@@ -936,6 +937,51 @@ Error MCCASBuilder::createStringSection(
   return Error::success();
 }
 
+/// Reads and returns the length field of a DWARF header contained in
+/// Reader, assuming Reader is positioned at the beginning of the section.
+/// The Reader's state is advanced to the first byte after the section.
+static Expected<uint64_t>
+getSizeFromDWARFHeaderAndSkip(BinaryStreamReader &SectionReader) {
+  // The DWARF headers have a standard format, if it is 32-bit DWARF, the first
+  // 4 bytes are a length field which we want to read and then skip over to get
+  // to the end of a function's contribution to the particular DWARF Section. If
+  // the first 4 bytes are 0xffffffff it is 64-bit DWARF and the next 8 bytes
+  // contain the actual length that we then skip.
```

Comment:
As discussed in the other PR, this is not an "if and only if statement".
The 64-bit format will use `0xffffffff`, but this doesn't imply that the 32-bit format uses _anything_ different from `0xffffffff`

Reply:
Yep, I hadn't realized that, I will make the change